tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/ZipUtil.java

```diff
@@ -0,0 +1,66 @@
+package com.linagora.tmail.james.jmap;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.james.jmap.routes.Blob;
+
+import reactor.core.publisher.Flux;
+
+public class ZipUtil {
+    public static Flux<byte[]> createZipStream(Collection<Blob> blobs) {
+        return Flux.create(sink -> {
+            try (ZipOutputStream zos = new ZipOutputStream(new OutputStream() {
```

Comment:
The override of the OutputSream to plug `OutputStream` into reactor-netty is very clever.

However like that it's really hard to read it and understand the intent: I had to got where the code is actually used!

I think we can actually improve this by extracting the class:

```
static class ReactorNettyOutputStreamAdapter extends OutputSream {
    private final Sink<...>

    // ...
}
```

Reply:
Not applied